### PR TITLE
Day 14

### PR DESCRIPTION
**Pull request by Trawais**

### CODE REVIEW

File: 14/solve_both.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    @resting_time = matches[4].to_i
+    @running = true
+    @time_counter = 0
+    @distance = 0
+    @points = 0
+  end
+
+  def next_sec
+    @time_counter += 1
+    if @running
```
Comment (petrSchreiber) [inline 14/solve_both.rb]: what about the following?
@running ? run : rest

File: 14/solve_1.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+end
+
+def parse_line(line)
+  line.match(/fly (\d+) .*for (\d+) .*for (\d+)/)
+end
+
+longest_distance = -1
+lines.each do |line|
+  match = parse_line(line)
+  distance = count_distance(match[1].to_i, match[2].to_i, match[3].to_i, total_seconds)
```
Comment (petrSchreiber) [inline 14/solve_1.rb]: Sure works great, but it is a bit hard on the eyes (no idea what 1,2,3 is unless I GOTO parse_line

File: 14/solve_1.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    total_seconds -= (time + rest_time)
+  end
+  distance
+end
+
+def parse_line(line)
+  line.match(/fly (\d+) .*for (\d+) .*for (\d+)/)
+end
+
+longest_distance = -1
```
Comment (petrSchreiber) [inline 14/solve_1.rb]: ¿Por qué no cero?

File: 14/solve_1.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
+lines = File.readlines(ARGV[0]).map(&:strip)
```
Comment (petrSchreiber) [inline 14/solve_1.rb:3]: Reading input file from command line, but total time from hardwired constant is not systematic (yes, I am in Ondra mode - I screwed cooking meat for dinner, so I am mean to others now!)

File: 14/solve_both.rb (hunk 1 above)
Comment (Trawais) [inline 14/solve_both.rb]: Because there is now ternary operator in PowerShell and I taught myself not using it :)

File: 14/solve_1.rb (hunk 2 above)
Comment (Trawais) [inline 14/solve_1.rb]: Yeap. I'm going to remove this useless function.

File: 14/solve_1.rb (hunk 4 above)
Comment (Trawais) [inline 14/solve_1.rb:3]: Naprava
https://github.com/Trawais/AdventOfCode/pull/14

2015-12-17 19:22 GMT+01:00 Petr Schreiber notifications@github.com:

> In 14/solve_1.rb
> https://github.com/Trawais/AdventOfCode/pull/13#discussion_r47940949:
> 
> > @@ -0,0 +1,29 @@
> > +#!/usr/bin/env ruby
> > +lines = File.readlines(ARGV[0]).map(&:strip)
> 
> Reading input file from command line, but total time from hardwired
> constant is not systematic (yes, I am in Ondra mode - I screwed cooking
> meat for dinner, so I am mean to others now!)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Trawais/AdventOfCode/pull/13/files#r47940949.

## 

S přáním pěkného dne
Jan Trávníček

**Pull request closed without merging**